haskell-src/Concordium/Utils/Serialization.hs

```diff
@@ -34,12 +36,22 @@ getLength = do
 putByteStringLen :: Putter BS.ByteString
 putByteStringLen bs = putLength (BS.length bs) >> putByteString bs
 
+-- |Put a 'LBS.ByteString' preceded by its length as a 64-bit (unsigned) integer.
+putLazyByteStringLen :: Putter LBS.ByteString
+putLazyByteStringLen bs = putLength (fromIntegral (LBS.length bs)) >> putLazyByteString bs
+
 -- |Get a 'BS.ByteString' preceded by its length as a 64-bit (unsigned) integer.
 getByteStringLen :: Get BS.ByteString
 getByteStringLen = do
     len <- getLength
     getByteString len
 
+-- |Get a 'LBS.ByteString' preceded by its length as a 64-bit (unsigned) integer.
+getLazyByteStringLen :: Get LBS.ByteString
```

Comment:
Note that this is no more efficient than `getByteStringLen` since the implementation is to get the bytestring and then convert it to a lazy one.

In particular this means that the underlying bytes are copied. Perhaps it would be good to document this caveat since in principle this could be different.

Reply:
Added documentation. The purpose of this function isn't to be more efficient, but to be more convenient. I think as a general rule, derserializers do not retain pointers to the underlying bytestrings, and that we should stick to this principle. Otherwise, this could lead to memory leaks where large bytestrings are maintained when only part of it is actually needed.